src/tribler-core/tribler_core/modules/metadata_store/community/tests/test_remote_query_community.py

```diff
@@ -200,3 +214,76 @@ def test_sanitize_query(self):
         ]
         for req, resp in req_response_list:
             self.assertDictEqual(sanitize_query(req), resp)
+
+    async def test_process_rpc_query_match_many(self):
+        """
+        Check if a correct query with a match in our database returns a result.
+        """
+        with db_session:
+            channel = self.channel_metadata(0).create_channel("a channel", "")
+            add_random_torrent(self.torrent_metadata(0), name="a torrent", channel=channel)
+
+        results = await self.overlay(0).process_rpc_query(dumps({}))
+        self.assertEqual(2, len(results))
+
+        channel_md, torrent_md = results if isinstance(results[0], self.channel_metadata(0)) else results[::-1]
+        self.assertEqual("a channel", channel_md.title)
+        self.assertEqual("a torrent", torrent_md.title)
+
+    async def test_process_rpc_query_match_one(self):
+        """
+        Check if a correct query with one match in our database returns one result.
+        """
+        with db_session:
+            self.channel_metadata(0).create_channel("a channel", "")
+
+        results = await self.nodes[0].overlay.process_rpc_query(dumps({}))
```

Comment:
Nitpicking: I seem to have forgotten porting some `self.nodes[0].overlay` statements to `self.overlay(0)`. Functionally equivalent and not much shorter.. but slightly easier on the eyes.

Reply:
Alright, nobody move: I'll fix this quickly.

Reply:
Done 😃